src/Option.ts

```diff
@@ -276,7 +276,7 @@ export function fromPredicate<A>(predicate: Predicate<A>): (a: A) => Option<A> {
 export function partitionMap<A, L, R>(f: (a: A) => Either<L, R>, fa: Option<A>): { left: Option<L>; right: Option<R> } {
   return fa.fold(
     () => ({ left: none, right: none }),
-    a => f(a).fold(l => ({ left: some(l), right: none }), a => ({ left: none, right: some(a) }))
+    a => f(a).fold(l => ({ left: some(l), right: none }), a => ({ left: none as Option<L>, right: some(a) }))
```

Comment:
because ts is using the left branch (`l => ({ left: some(l), right: none })`) to infer the types, I pushed a cleaner solution